sql/migrate/lex.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2021-present The Atlas Authors. All rights reserved.
+// This source code is licensed under the Apache 2.0 license found
+// in the LICENSE file in the root directory of this source tree.
+
+package migrate
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+	"unicode/utf8"
+)
+
+// stmts provides a generic implementation for extracting
+// SQL statements from the given file string.
+func stmts(input string) ([]string, error) {
```

Comment:
Can we change this to an `io.Reader`? This feels very un-goish.

Reply:
not really. I'm using the `strings` functions and Reader will make my work harder. Also, parsers usually work on bytes or strings.